From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Tue, 24 Jun 2014 15:03:21 +0200
Subject: [129/249] staging: rtl8723au: rtw_check_beacon_data23a(): Capability
 is already uptodate in cur_network
Origin: https://git.kernel.org/linus/51a26d26feec3e8d41092f33fde6b111961d59b2

Beacon capability was already added to mlme_priv->cur_network before
calling this function, so no need to retrieve it from the IE array
again.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/core/rtw_ap.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/rtl8723au/core/rtw_ap.c b/drivers/staging/rtl8723au/core/rtw_ap.c
index 08e933a..c19ed95 100644
--- a/drivers/staging/rtl8723au/core/rtw_ap.c
+++ b/drivers/staging/rtl8723au/core/rtw_ap.c
@@ -797,7 +797,7 @@ int rtw_check_beacon_data23a(struct rtw_adapter *padapter,
 	u8 *pHT_caps_ie = NULL;
 	u8 *pHT_info_ie = NULL;
 	struct sta_info *psta = NULL;
-	u16 cap, ht_cap = false;
+	u16 ht_cap = false;
 	uint ie_len = 0;
 	int group_cipher, pairwise_cipher;
 	u8 channel, network_type, supportRate[NDIS_802_11_LENGTH_RATES_EX];
@@ -844,9 +844,6 @@ int rtw_check_beacon_data23a(struct rtw_adapter *padapter,
 
 	memcpy(pbss_network->MacAddress, myid(&padapter->eeprompriv), ETH_ALEN);
 
-	/* capability */
-	cap = get_unaligned_le16(ie);
-
 	/* SSID */
 	p = rtw_get_ie23a(ie + _BEACON_IE_OFFSET_, WLAN_EID_SSID, &ie_len,
 			  (pbss_network->IELength -_BEACON_IE_OFFSET_));
@@ -894,7 +891,7 @@ int rtw_check_beacon_data23a(struct rtw_adapter *padapter,
 		ERP_IE_handler23a(padapter, p);
 
 	/* update privacy/security */
-	if (cap & BIT(4))
+	if (pbss_network->capability & BIT(4))
 		pbss_network->Privacy = 1;
 	else
 		pbss_network->Privacy = 0;
